packages/ses/src/lockdown.js

```diff
@@ -115,20 +118,43 @@ const assertDirectEvalAvailable = () => {
     // and indirect, which generally creates a new global.
     // We are going to throw an exception for failing to initialize SES, but
     // good neighbors clean up.
-    if (!allowed) {
+    if (!evalAllowed) {
       delete globalThis.SES_changed;
     }
   } catch (_error) {
     // We reach here if eval is outright forbidden by a Content Security Policy.
     // We allow this for SES usage that delegates the responsibility to isolate
     // guest code to production code generation.
-    allowed = true;
+    evaluatorsSuccessful = false;
   }
-  if (!allowed) {
-    // See https://github.com/endojs/endo/blob/master/packages/ses/error-codes/SES_DIRECT_EVAL.md
-    throw TypeError(
-      `SES cannot initialize unless 'eval' is the original intrinsic 'eval', suitable for direct-eval (dynamically scoped eval) (SES_DIRECT_EVAL)`,
-    );
+
+  switch (hostEvaluators) {
+    case 'all':
+      assert(
+        evalAllowed === true &&
+          functionAllowed === true &&
+          evaluatorsSuccessful === true,
+        "SES cannot initialize unless 'eval' is the original intrinsic 'eval', suitable for direct-eval (dynamically scoped eval) (SES_DIRECT_EVAL)",
+      );
+      break;
+    case 'none':
+      assert(
+        evalAllowed === false &&
+          functionAllowed === false &&
+          evaluatorsSuccessful === false,
+        '"hostEvaluators" was set to "none", but evaluators are not blocked (SES_DIRECT_EVAL)',
+      );
+      break;
+    case 'no-direct':
+      assert(
+        evalAllowed === false &&
+          functionAllowed === true &&
+          evaluatorsSuccessful === true,
+        `"hostEvaluators" was set to "no-direct", but ${evalAllowed === true ? 'direct eval is functional' : 'evaluators are not allowed'} (SES_DIRECT_EVAL)`,
+      );
+      break;
+    default:
+      throw TypeError('Invalid hostEvaluators option');
```

Comment:
```suggestion
      throw TypeError(`Invalid hostEvaluators option ${hostEvaluators}`);
```

Reply:
```suggestion
      Fail`Invalid hostEvaluators option ${hostEvaluators}`;
```

Reply:
or remove since never reached

we reach here first

https://github.com/endojs/endo/blob/13989af9fd1206f95659a7fd779b73ad12c243fb/packages/ses/src/lockdown.js#L233-L236